src/credentials-manager/index.js

```diff
@@ -0,0 +1,145 @@
+import {NativeModules} from 'react-native';
+import CredentialsManagerError from './credentialsManagerError';
+
+export default class CredentialsManager {
+  constructor(clientId, domain) {
+    this.domain = domain;
+    this.clientId = clientId;
+    this.Auth0Module = NativeModules.A0Auth0;
+  }
+
+  /**
+   * Saves the provided credentials
+   *
+   * @param {Object} credentials credential values
+   * @param {String} credentials.idToken required - JWT token that has user claims
+   * @param {String} credentials.accessToken required - token used for API calls
+   * @param {String} credentials.tokenType required - type of the token, ex - Bearer
+   * @param {Number} credentials.expiresIn optional - `expiresAt` should not be empty if this is. Used to denote when the token will expire from the issued time
+   * @param {String} credentials.expiresAt optional - `expiresIn` should not be empty if this is. Used to denote when the token will expire. Has precendence over `expiresIn`.
+   * @param {String} credentials.refreshToken optional - used to refresh access token
+   * @param {String} credentials.scope optional - represents the scope of the current token
+   * @returns {Promise}
+   *
+   * @memberof CredentialsManager
+   */
+  async saveCredentials(credentials = {}) {
+    const validateKeys = ['idToken', 'accessToken', 'tokenType'];
+    validateKeys.forEach(key => {
+      if (!credentials[key]) {
+        const json = {
+          error: 'a0.credential_manager.invalid_input',
+          error_description: `${key} cannot be empty`,
+          invalid_parameter: key,
+        };
+        throw new CredentialsManagerError({json, status: 0});
+      }
+    });
+    if (
+      (!credentials.expiresIn || !credentials.expiresIn > 0) &&
+      !credentials.expiresAt
+    ) {
+      const json = {
+        error: 'a0.credential_manager.invalid_input',
+        error_description: `expiresIn or expiresAt should be set`,
+      };
+      throw new CredentialsManagerError({json, status: 0});
+    }
+    try {
+      await this.ensureCredentialManagerIsInitialized();
+      await this.Auth0Module.saveCredentials(credentials);
+    } catch (e) {
+      const json = {
+        error: 'a0.credential_manager.invalid',
+        error_description: e.message,
+      };
+      throw new CredentialsManagerError({json, status: 0});
+    }
+  }
+
+  /**
+   * Gets the credential that has already been saved
```

Comment:
We're using "credentials" vs "credential" most other places, we should be consistent.

```suggestion
   * Gets the credentials that has already been saved
```